tests/functional/stores/client.rs

```diff
@@ -188,6 +188,70 @@ async fn client_upsert_id(ctx: &mut StoreContext) {
         .unwrap();
 }
 
+#[test_context(StoreContext)]
+#[tokio::test]
+async fn client_create_same_id_and_token(ctx: &mut StoreContext) {
+    let client_id = format!("id-{}", gen_id());
+    let token = format!("token-{}", gen_id());
+
+    // Initial Client creation
+    ctx.clients
+        .create_client(TENANT_ID, &client_id, Client {
+            tenant_id: TENANT_ID.to_string(),
+            push_type: ProviderKind::Fcm,
+            token: token.clone(),
+        })
+        .await
+        .unwrap();
+    let insert_result = ctx.clients.get_client(TENANT_ID, &client_id).await.unwrap();
+    assert_eq!(insert_result.token, token.clone());
+
+    // Insert notification for the client to test the clients->notifications
+    // constraint works properly
+    let notification_id = format!("id-{}", gen_id());
+    let notification_payload = format!("payload-{}", gen_id());
+
+    ctx.notifications
+        .create_or_update_notification(
+            &notification_id,
+            TENANT_ID,
+            &client_id,
+            &echo_server::handlers::push_message::MessagePayload {
+                topic: String::new(),
+                flags: 0,
+                blob: notification_payload,
+            },
+        )
+        .await
+        .unwrap();
+    let get_notification_result = ctx
+        .notifications
+        .get_notification(&notification_id, TENANT_ID)
+        .await
+        .unwrap();
+    assert_eq!(get_notification_result.client_id, client_id);
+
+    // Create a Client with the same id and same device token
+    // changing only the push_type to check if it was upserted
+    ctx.clients
+        .create_client(TENANT_ID, &client_id, Client {
+            tenant_id: TENANT_ID.to_string(),
+            push_type: ProviderKind::Noop,
+            token: token.clone(),
+        })
+        .await
+        .unwrap();
+    let double_insert_result = ctx.clients.get_client(TENANT_ID, &client_id).await.unwrap();
+    assert_eq!(double_insert_result.token, token);
+    assert_eq!(double_insert_result.push_type, ProviderKind::Noop);
+
```

Comment:
Can we add (perhaps in another PR) a test that the notification still exists in the DB for this client? Also same for other tests where the client ID is the same

Reply:
I think this case should not happen as we have [ON DELETE CASCADE in the notifications table](https://github.com/WalletConnect/echo-server/blob/3dde225228178c6198c89d674113c05ad4eb700b/migrations/1695381202_alter_notifications_constraint.sql#L6): notifications should be removed when the related client->id is removed.

Reply:
Yeah but I'm saying if the client_id is re-added, the notifications shouldn't be deleted.

Reply:
Created #236 to track this comment.